core/io/stream_peer.cpp

```diff
@@ -492,6 +492,8 @@ Error StreamPeerBuffer::get_partial_data(uint8_t* p_buffer, int p_bytes,int &r_r
 
 	PoolVector<uint8_t>::Read r = data.read();
 	copymem(p_buffer,r.ptr(),r_received);
+	
+	// FIXME: return what? OK or ERR_*
```

Comment:
It looks like the error handling is actually done in `get_data`, where it returns `ERR_INVALID_PARAMETER` if `p_bytes != r_received`. So maybe this method here should be changed to return `void`, or the error handling should be added directly here and `get_data` should just return `get_partial_data(p_buffer,p_bytes,recv)`.

@Faless @punto- Any idea?